File: app/src/main/java/com/ptithcm/thuan6420/basecleanarchitecture/ui/components/UiController.java
package com.ptithcm.thuan6420.basecleanarchitecture.ui.components;

import android.app.Activity;
import android.app.ProgressDialog;
import android.content.Context;
import android.view.View;
import android.view.inputmethod.InputMethodManager;

import com.ptithcm.thuan6420.basecleanarchitecture.ui.util.ConstantMessage;

public class UiController {
    public static UiController Instance;
    private static ProgressDialog mProgressDialog;

    public void initProgressDialog(Context context) {
        mProgressDialog = new ProgressDialog(context);
        mProgressDialog.setCancelable(false);
        mProgressDialog.setMessage(ConstantMessage.MESSAGE_LOADING);
    }

    public static UiController getInstance() {
        if (Instance != null) {
            return Instance;
        }
        return new UiController();
    }

    public void closeKeyBoard(Activity activity) {
        View view = activity.getCurrentFocus();
        if (view == null) {
            return;
        }
        InputMethodManager inputMethodManager = (InputMethodManager) activity
                .getSystemService(Context.INPUT_METHOD_SERVICE);
        inputMethodManager.hideSoftInputFromWindow(view.getWindowToken(), 0);
    }

    public void turnOnLoading(Context context) {
        mProgressDialog.show();
    }

    public void turnOffLoading() {
       mProgressDialog.dismiss();
    }

}

File: app/src/main/java/com/ptithcm/thuan6420/basecleanarchitecture/ui/components/UiController.kt
package com.ptithcm.thuan6420.basecleanarchitecture.ui.components

import android.app.Activity

import android.content.Context
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.ProgressBar
import com.ptithcm.thuan6420.basecleanarchitecture.ui.util.ConstantMessage

class UiController {
    private var mProgressBar: ProgressBar? = null

    fun initProgressDialog(context: Context?) {
        mProgressBar = ProgressBar(context)
    }

    fun closeKeyBoard(activity: Activity) {
        val view = activity.currentFocus ?: return
        val inputMethodManager = activity
            .getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        inputMethodManager.hideSoftInputFromWindow(view.windowToken, 0)
    }

    fun turnOnLoading(context: Context?) {
        mProgressBar?.visibility = View.VISIBLE
    }

    fun turnOffLoading() {
        mProgressBar?.visibility = View.GONE
    }

    companion object{
        private var Instance : UiController? = null
        operator fun invoke() = synchronized(this) {
            if (Instance != null) {
                Instance
            }
            Instance = UiController()
        }
    }
}